### PR TITLE
Api1600 storage

### PR DESCRIPTION
**Pull request by nabhajit-ray**
### Description
[Api1600 support for storage]

### Issues Resolved
[]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: libraries/resource_providers/api500/c7000/storage_pool_provider.rb [hunk 1]
```diff
@@ -61,6 +61,7 @@ def remove
         def update
           load_storage_system
           desired_state = Marshal.load(Marshal.dump(@item.data))
+          desired_state.delete('type')
```
Comment (AsisBagga) [inline libraries/resource_providers/api500/c7000/storage_pool_provider.rb:64]: any reason why are we deleting type key value pair?
I hope while retrieving the existing resource, type should be compared isn't it?

Comment (nabhajit-ray) [reply libraries/resource_providers/api500/c7000/storage_pool_provider.rb:64]: The API docs says we dont need type field while doing update and it complains if we have that in our JSON body

**Review by AsisBagga — APPROVED** [review]
LGTM

**Review by VenkateshRavula — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,10 +13,16 @@
...
 - oneview_server_hardware_type
+- oneview_server_profile
+- oneview_server_profile_template
 - oneview_storage_pool
 - oneview_storage_system
+- oneview_storage_volume
 - oneview_volume_template
 
+Adds API1600 support to HPE Synergy Image Streamer resources:
+  - image_streamer_deployment_plan
```
Comment (VenkateshRavula) [inline CHANGELOG.md:24]: if we are listing image streamer resource separately, can you add the artifact bundles resource also in the same way.

**Review by VenkateshRavula — APPROVED** [review]

**Review by chebroluharika — APPROVED** [review]

**Pull request merged**